src/foundry/common/abstract/document.d.mts

```diff
@@ -2007,39 +2011,76 @@ declare namespace Document {
         parent: Parent;
       };
 
-  type DefaultNameContext<SubType extends string, Parent extends Document.Any | null> = {
-    /** The sub-type of the document */
-    type?: SubType | undefined;
-
-    /** A compendium pack within which the Document should be created */
-    pack?: string | undefined;
-  } & ParentContext<Parent>;
+  /** @internal */
+  type _PossibleSubtypeContext<DocumentName extends Document.Type> =
+    GetKey<Document.MetadataFor<DocumentName>, "hasTypeData"> extends true
```

Comment:
While it's fine in this case, I will note that this doesn't distribute and so would normally be a logical issue. Maybe you already were aware of that in case great!